nogo is very noisy in 0.50.0

### What version of rules_go/gazelle/Bazel are you using?

rules_go 0.50.0, gazelle 0.38.0, Bazel 7.3.1, Go 1.23.0

### What did you do?

I wrote some Go code that doesn't compile, and then I tried to build it with `bazel build //...`

### What did you expect to see?

An error message

### What did you see instead?

28 error messages.

Each nogo analyzer is individually reporting on the compilation failure. For example:

    analyzer "asmdecl" failed: analysis skipped due to type-checking error: path/to/file.go:142:71: cannot use b.awsConfig (variable of type *aws.Config) as aws.Config value in argument to s3.NewFromConfig

... repeated for every nogo analyzer.

I can't seem to reproduce this with _every_ compilation failure, which is interesting. The line that manages to trigger this is `s3.NewFromConfig(b.awsConfig)` from `github.com/aws/aws-sdk-go-v2/service/s3`, where `b.awsConfig` is a pointer instead of a struct.

--- a/go/tools/builders/nogo_main.go
+++ b/go/tools/builders/nogo_main.go
@@ -372,9 +372,7 @@ func (act *action) execOnce() {
 	act.pass = pass
 
 	var err error
-	if act.pkg.illTyped && !pass.Analyzer.RunDespiteErrors {
-		err = fmt.Errorf("analysis skipped due to type-checking error: %v", act.pkg.typeCheckError)
-	} else {
+	if !act.pkg.illTyped || pass.Analyzer.RunDespiteErrors {
 		act.result, err = pass.Analyzer.Run(pass)
 		if err == nil {
 			if got, want := reflect.TypeOf(act.result), pass.Analyzer.ResultType; got != want {
@@ -471,7 +469,13 @@ func checkAnalysisResults(actions []*action, pkg *goPackage) string {
 	if cwd == "" || err != nil {
 		errs = append(errs, fmt.Errorf("nogo failed to get CWD: %w", err))
 	}
+	numSkipped := 0
 	for _, act := range actions {
+		if act.pkg.illTyped && !act.a.RunDespiteErrors {
+			// Don't report type-checking errors once per analyzer.
+			numSkipped++
+			continue
+		}
 		if act.err != nil {
 			// Analyzer failed.
 			errs = append(errs, fmt.Errorf("analyzer %q failed: %v", act.a.Name, act.err))
@@ -542,6 +546,9 @@ func checkAnalysisResults(actions []*action, pkg *goPackage) string {
 				diagnostics = append(diagnostics, entry{Diagnostic: d, Analyzer: act.a})
 			}
 		}
+	}
+	if numSkipped > 0 {
+		errs = append(errs, fmt.Errorf("%d analyzers skipped due to type-checking error: %v", numSkipped, pkg.typeCheckError))
 	}
 	if len(diagnostics) == 0 && len(errs) == 0 {
 		return ""

--- a/tests/core/nogo/custom/custom_test.go
+++ b/tests/core/nogo/custom/custom_test.go
@@ -31,7 +31,7 @@ func TestMain(m *testing.M) {
 		Nogo: "@//:nogo",
 		Main: `
 -- BUILD.bazel --
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "nogo")
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library", "nogo")
 
 nogo(
     name = "nogo",
@@ -122,6 +122,12 @@ go_library(
     name = "dep",
     srcs = ["dep.go"],
     importpath = "dep",
+)
+
+go_binary(
+	name = "type_check_fail",
+	srcs = ["type_check_fail.go"],
+	pure = "on",
 )
 
 -- foofuncname.go --
@@ -434,6 +440,16 @@ func Foo() bool { // This should fail foofuncname
   return Bar()
 }
 
+-- type_check_fail.go --
+package type_check_fail
+
+import (
+	"strings"
+)
+
+func Foo() bool {
+  return strings.Split("a,b,c", ",") // This fails type-checking
+}
 `,
 	})
 }
@@ -443,6 +459,7 @@ func Test(t *testing.T) {
 		desc, config, target string
 		wantSuccess          bool
 		includes, excludes   []string
+		bazelArgs []string
 	}{
 		{
 			desc:        "default_config",
@@ -508,6 +525,16 @@ func Test(t *testing.T) {
 				`examplepkg[\\/]pure_src_with_err_calling_native.go:.*function must not be named Foo \(foofuncname\)`,
 			},
 		}, {
+			desc:        "type_check_fail",
+			config:      "config.json",
+			target:      "//:type_check_fail",
+			wantSuccess: false,
+			includes: []string{
+				"3 analyzers skipped due to type-checking error: type_check_fail.go:8:10:",
+			},
+			// Ensure that nogo runs even though compilation fails
+			bazelArgs: []string{"--keep_going"},
+		}, {
 			desc:        "no_errors",
 			target:      "//:no_errors",
 			wantSuccess: true,
@@ -527,7 +554,7 @@ func Test(t *testing.T) {
 				defer replaceInFile("BUILD.bazel", customConfig, origConfig)
 			}
 
-			cmd := bazel_testing.BazelCmd("build", test.target)
+			cmd := bazel_testing.BazelCmd(append([]string{"build", test.target}, test.bazelArgs...)...)
 			stderr := &bytes.Buffer{}
 			cmd.Stderr = stderr
 			if err := cmd.Run(); err == nil && !test.wantSuccess {

--- a/tests/core/nogo/custom/custom_test.go
+++ b/tests/core/nogo/custom/custom_test.go
@@ -530,7 +530,7 @@ func Test(t *testing.T) {
 			target:      "//:type_check_fail",
 			wantSuccess: false,
 			includes: []string{
-				"3 analyzers skipped due to type-checking error: type_check_fail.go:8:10:",
+				"4 analyzers skipped due to type-checking error: type_check_fail.go:8:10:",
 			},
 			// Ensure that nogo runs even though compilation fails
 			bazelArgs: []string{"--keep_going"},

